### ISSUE TITLE
SDRi cannot reach 19.1db

### ISSUE THREAD

**Issue by Tinglok** [ShiZiqiang/dual-path-RNNs-DPRNNs-based-speech-separation#2]
Thanks for your contribution, which helps me a lot. However, when I tried to train the model by myself and used the evaluation script that Kaituo proposed in [Conv-TasNet](https://github.com/kaituoxu/Conv-TasNet/blob/master/src/evaluate.py), I couldn't reach the evaluation metric as proposed (my SDRi is 13.43). Even I used your pretrained model for evaluation, I still couldn't get the proposed SDRi (I got 14.97).

During the training stage, something strange happened when I trained about 60 epoch. The loss suddenly changed to 80 and never changes again, which may be caused by the gradient vanish problem.

Could you please help me solve this problem? thanks again.

**Comment by ShiZiqiang** [comment]
Hi, Tinglok. Sorry for late reply. Thanks for your message. I think there are two questions in the message. 

1. cannot reach the target SDRi.  

You should check the WSJ-2mix data you generated from WSJ0. Since there are utterances of the same name but recorded under different conditions. So you should mix the right file of the same name to generate the WSJ-2mix data.

2. Suddenly drop of the SDR to 80.

Yes. We have met the same problem. Just restart the training from the last best checkpoint, and half the learning rate to continue the training.

If you have any problem. Please feel free to contact.

**Comment by Tinglok** [comment]
Hi Ziqiang,

Thanks for you reply. I am regenerating the WSJ-2mix data right now, but I was wondering could you please provide the json file that generated in step 1, which might be useful for me because it has wav length. 

If it's not convenient, my [json file](https://github.com/ShiZiqiang/dual-path-RNNs-DPRNNs-based-speech-separation/files/4168928/json.zip) in tr, cv and tt folder are attached, please help me find out if there is something wrong.

Thanks!

**Comment by ShiZiqiang** [comment]
Hi, Tinglok,

Please find my json file 
[data.zip](https://github.com/ShiZiqiang/dual-path-RNNs-DPRNNs-based-speech-separation/files/4169829/data.zip)
and check it.

I use bss_eval_sources to evaluate the performance to get the SDR. I think this  bss_eval_sources is standard. And get from the original MERL page.

**Comment by Tinglok** [comment]
Hi Ziqiang,

Thanks for your quick response! After checking, I found that the json file you provided are the same as mine. Also, I regenerate the data and found it doesn't work. 

Could you please tell me your package version, especially torch, mir_eval, and librosa. This could be the last thing I think may go wrong.😥

Thanks for your time~

**Comment by Tinglok** [comment]
Hi Ziqiang,

Finally, I solved this bug. The problem is really as you said, that there are two formats: wv1 and wv2, where wv1 is clean and wv2 is noisy.

Anyway, thanks for your kindness, which help me find the bug and fixed it finally.
